flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -161,15 +169,37 @@ public static void setupYarnClassPath(Configuration conf, Map<String, String> ap
 
 		fs.copyFromLocalFile(false, true, localSrcPath, dst);
 
-		// Note: If we used registerLocalResource(FileSystem, Path) here, we would access the remote
+		// Note: If we directly used registerLocalResource(FileSystem, Path) here, we would access the remote
 		//       file once again which has problems with eventually consistent read-after-write file
-		//       systems. Instead, we decide to preserve the modification time at the remote
-		//       location because this and the size of the resource will be checked by YARN based on
-		//       the values we provide to #registerLocalResource() below.
-		fs.setTimes(dst, localFile.lastModified(), -1);
-		// now create the resource instance
-		LocalResource resource = registerLocalResource(dst, localFile.length(), localFile.lastModified());
```

Comment:
I think we no longer need `registerLocalResource(Path, long, long)`

Reply:
we still do below:
```
LocalResource resource = registerLocalResource(dst, localFile.length(), dstModificationTime > 0 ? dstModificationTime
```

Reply:
Good point. I overlooked this. Please ignore this comment.